plugins/synapticsmst/synapticsmst-device.c

```diff
@@ -247,24 +247,26 @@ synapticsmst_device_scan_cascade_device (SynapticsMSTDevice *device,
 	guint8 byte[4];
 	g_autoptr(SynapticsMSTConnection) connection = NULL;
 	g_autoptr(GError) error_local = NULL;
+	g_autoptr(FuDeviceLocker) locker = NULL;
 
 	if (priv->test_mode)
 		return TRUE;
 
 	/* reset */
 	priv->has_cascade = FALSE;
 
-	if (!synapticsmst_device_enable_remote_control (device, error)) {
-		g_prefix_error (error,
-				"failed to enable remote control on tx_port %d: ",
-				tx_port);
+	/* enable remote control and disable on exit */
+	locker = fu_device_locker_new_full (device,
+					    (FuDeviceLockerFunc) synapticsmst_device_enable_remote_control,
+					    (FuDeviceLockerFunc) synapticsmst_device_disable_remote_control,
```

Comment:
I think this is exactly how you should use FuDeviceLocker -- much appreciated. Why not just call it `synapticsmst_device_open()` and `synapticsmst_device_close()`? Then it's very obvious what it's doing.

Reply:
Actually it's not needed for all communication requests, only specific ones.  These commits are the basis to allow code reuse across plugins (symlink the source files and compile into both plugins) and those may not necessarily use remote control functionality (TBD).

Reply:
Ahh okay, NP then.